---
title: C6310 | Microsoft Docs
ms.date: 11/15/2016
ms.prod: visual-studio-dev14
ms.technology: vs-ide-code-analysis
ms.topic: reference
f1_keywords:
- C6310
helpviewer_keywords:
- C6310
ms.assetid: e759eb63-883f-4c3e-bf2f-b924ff511405
caps.latest.revision: 15
author: corob-msft
ms.author: corob
manager: jillfra
ms.openlocfilehash: a04a37e0c95915849e46bcf3ffd74f8afff2cb56
ms.sourcegitcommit: 6cfffa72af599a9d667249caaaa411bb28ea69fd
ms.translationtype: MT
ms.contentlocale: tr-TR
ms.lasthandoff: 09/02/2020
ms.locfileid: "77273804"
---
# <a name="c6310"></a>C6310
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

Uyarı C6310: özel durum filtresinde geçersiz sabit beklenmeyen davranışlara neden olabilir  
  
 Bu ileti, yapılandırılmış bir özel durum işleyicisinin filtre ifadesinde geçersiz bir sabit olduğunu gösterir. Yapılandırılmış bir özel durum işleyicisinin filtre ifadesinde kullanılmak üzere tanımlanan sabitler şunlardır:  
  
- `EXCEPTION_CONTINUE_EXECUTION`  
  
- `EXCEPTION_CONTINUE_SEARCH`  
  
- `EXCEPTION_EXECUTE_HANDLER`  
  
  Bu değerler, excpt. h çalışma zamanı başlık dosyasında tanımlanmıştır.  
  
  Önceki listede olmayan bir sabit kullanmak beklenmeyen davranışlara neden olabilir.  
  
## <a name="example"></a>Örnek  
 Aşağıdaki kod bu uyarıyı oluşturur:  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
BOOL LimitExceeded();  
  
void fd( )  
{  
  __try   
  {  
    if (LimitExceeded())   
    {  
      RaiseException(EXCEPTION_ACCESS_VIOLATION,0,0,0);  
    }  
    else  
    {  
      // code   
    }  
  }   
  __except (EXCEPTION_ACCESS_VIOLATION)  
  {  
    puts("Exception Occurred");  
  }  
}  
```  
  
 Bu uyarıyı düzeltmek için aşağıdaki kodu kullanın:  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
BOOL LimitExceeded();  
  
void fd( )  
{  
  __try   
  {  
    if (LimitExceeded())   
    {  
      RaiseException(EXCEPTION_ACCESS_VIOLATION,0,0,0);  
    }  
    else  
    {  
      // code   
    }  
  }   
  __except (GetExceptionCode()==EXCEPTION_ACCESS_VIOLATION ?   
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)  
  {  
    puts("Exception Occurred");  
  }  
}  
```
